modules/core/src/lib/layer.ts

```diff
@@ -434,11 +435,39 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
 
   // Default implementation
   // Sublayers chould override this method to provide an accurate calculation of the bounds
-  getBounds(): [number[], number[]] | null {
+  getBounds(attributes?: string[]): [number[], number[]] | null {
```

Comment:
This is a fairly big chunk of independent logic to push into the already ripe-for-refactor-and-cleanup `Layer` class. Have we looked at other places where some of this could go? 

It seems intimately related to attributes, AttributeManager is not the right place?

Reply:
I thought that too, but I think it is too much for this PR. @Pessimistress do you have thoughts on moving it to `AttributeManager`?